components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -164,6 +164,13 @@ private void processFido2AuthenticationResponse(AuthenticatedUser user, String t
                 tokenResponse);
     }
 
+    private AuthenticatedUser processFido2UsernamelessAuthenticationResponse(String tokenResponse)
+            throws AuthenticationFailedException {
+
+        WebAuthnService webAuthnService = new WebAuthnService();
```

Comment:
Isn't this an OSGI service. Why can't we make it so

Reply:
WebAuthnService is not currently implemented as an OSGi service. However, +1 to make it available as an OSGi service.

Reply:
This will be addressed with https://github.com/wso2/product-is/issues/7132